libyul/Object.cpp

```diff
@@ -65,9 +65,45 @@ set<YulString> Object::dataNames() const
 {
 	set<YulString> names;
 	names.insert(name);
```

Comment:
https://github.com/ethereum/solidity/pull/9047#discussion_r431791848

Object names can't contain dots, right? I'd add an assertion here for that.

And if they can, please add a test with such a name (unless one already exists).